chain/network/tests/routing.rs

```diff
@@ -287,11 +326,32 @@ struct Runner {
     num_validators: usize,
     peer_max_count: u32,
     state_machine: Option<StateMachine>,
+    /// If v is in the list from u, it means v is blacklisted from u point of view.
+    /// blacklist[u].contains(v) <=> v is blacklisted from u.
+    blacklist: HashMap<usize, HashSet<Option<usize>>>,
```

Comment:
Why do you need `HashSet<Option<usize>>` here?

Reply:
I use None in this test to blacklist all peers at once. It blacklist `127.0.0.1` (by IP instead of by IP:Port pair). Main goal of this is test that blacklist by IP works fine, will add comment about this.